### PR TITLE
Enable arbitrary profile in standard compliance scan

### PR DESCRIPTION
**Pull request by jan-cerny**
Adds a new option '--profile' to 'oscapd-evaluate scan' command.
This allows us to check compliance of given target to any security
standard. The profile needs to be available in SCAP Security guide
for the target platform.

### CODE REVIEW

**Comment by jan-cerny** [comment]
It looks like that Fedora 24 node on Jenkins hasn't `python-dbus` installed.

**Comment by jan-cerny** [comment]
I have installed python-dbus and python3-dbus where it was missing.

@openscap-jenkins test this please

**Comment by jan-cerny** [comment]
🤒 OK, so `gobject` is missing also on F24.

**Comment by jan-cerny** [comment]
@openscap-jenkins test this please

File: bin/oscapd-evaluate [hunk 1]
```diff
@@ -182,7 +183,7 @@ def cli_scan(args, config):
                         except:
                             logging.exception(
                                 "Failed to scan target '%s' for "
-                                "standard profile compliance." % (target)
+                                "standard compliance." % (target)
```
Comment (mpreisler) [inline bin/oscapd-evaluate]: change this to "configuration compliance" since it's no longer standard profile

File: bin/oscapd-evaluate [hunk 2]
```diff
@@ -467,14 +469,20 @@ def main():
...
+        help="Specify the profile ID for standard compliance scan. "
+        "If not specified, the 'standard' profile will be used."
     )
     scan_parser.add_argument(
         "--output", type=str, required=True,
         help="A directory where results will be stored in. There will be a "
         "directory for each target created there with up to 3 files. 'json' "
         "with json summary of the scan, cve.xml with CVE scan raw results and "
-        "std.xml with standard profile scan raw results."
+        "std.xml with standard compliance scan raw results."
```
Comment (mpreisler) [inline bin/oscapd-evaluate]: this too

**Comment by jan-cerny** [comment]
I have replaced "standard compliance" with "configuration compliance".

File: openscap_daemon/cli_helpers.py [hunk 3]
```diff
@@ -263,8 +263,8 @@ def summarize_standard_compliance_results(arf_source, result_list):
 
     test_result = arf_root.find(
         ".//cdf:TestResult[@id='%s']" %
-        ("xccdf_org.open-scap_testresult_xccdf_org.ssgproject.content_profile_"
-         "standard"), namespaces
+        # this ID prefix is hardcoded in oscap
+        ("xccdf_org.open-scap_testresult_" + profile), namespaces
```
Comment (mpreisler) [inline openscap_daemon/cli_helpers.py]: The prefix is SSG specific, not hardcoded in oscap

Comment (jan-cerny) [inline openscap_daemon/cli_helpers.py]: @mpreisler I don't think so. This prefix is part of TestResult@id of every XCCDF results that is created by OpenSCAP. I think it isn't SSG specific. If some content that isn't from SSG is evaluated, the XCCDF results contain this prefix also. Actually, what was here before was SSG specific. 

You can find the string in https://github.com/OpenSCAP/openscap/blob/maint-1.2/src/XCCDF_POLICY/xccdf_policy.c#L2011

Anyway, it's just a comment, I wrote it there to explain the "magic constant", but there's no problem to remove the comment.

Comment (mpreisler) [inline openscap_daemon/cli_helpers.py]: Yeah, you are right.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**